packages/react-native-reanimated/src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -558,18 +569,37 @@ export function createAnimatedComponent(
           if (sharedTransitionTag) {
             this._configureSharedTransition();
           }
-
-          const skipEntering = this.context?.current;
-          if (entering && !skipEntering) {
-            updateLayoutAnimations(
-              tag as number,
-              LayoutAnimationType.ENTERING,
-              maybeBuild(
-                entering,
-                this.props?.style,
-                AnimatedComponent.displayName
-              )
+          if (sharedTransitionTag && !IS_WEB) {
+            const sharedElementTransition =
+              this.props.sharedTransitionStyle ?? new SharedTransition();
+            const reduceMotionInTransition = getReduceMotionFromConfig(
+              sharedElementTransition.getReduceMotion()
             );
+            if (!reduceMotionInTransition) {
+              sharedElementTransition.registerTransition(
```

Comment:
`registerTransition` seems to already check for Reduced Motion, and +1 with piaskowyk here, let's try to move this logic to the proper manager code.

Reply:
`this._configureSharedTransition()` already does all of that, I think I  accidentally change this logic while experimenting